include/sys/arc_impl.h

```diff
@@ -264,9 +264,11 @@ typedef struct l2arc_log_ent_phys {
 /*
  * Maximum amount of data in an l2arc log block (used to terminate rebuilding
  * before we hit the write head and restore potentially corrupted blocks).
+ * Since the max recordsize/volblocksize is 128Kbytes, the max payload would
+ * be 128Kbytes * 1023 entries.
  */
 #define	L2ARC_LOG_BLK_MAX_PAYLOAD_SIZE	\
-	(L2ARC_LOG_BLK_SIZE * L2ARC_LOG_BLK_ENTRIES)
+	(128 * 1024 * L2ARC_LOG_BLK_ENTRIES)
 /*
```

Comment:
All fine and dandy, but this shows 1024, not 1023, which is the richt one?

Reply:
I think the code is clear here: L2ARC_LOG_BLK_ENTRIES = 1023, 128*1024=128KBytes. So 1023 entries, each with 128Kbytes data.

Reply:
The L2ARC_LOG_BLK_SIZE is defined to be 128Kbytes but that refers to the log block size (not to the data in each entry), which consists of 1023 entries, each with 128 bytes metadata plus the header of the log block size which is 128 bytes itself.
I thought it misleading as it was previously, but perhaps I am wrong.

Reply:
SHOOT, My mistake sorry... Clearly didn't had my coffee yet... >.<